api/v1_create_reward_code.go

```diff
@@ -90,17 +97,74 @@ func (app *ApiServer) v1CreateRewardCode(c *fiber.Ctx) error {
 		return err
 	}
 
-	// Verify the signature against authorized keys from config
-	matchedKey, err := verifySignatureAgainstKeys(req.Signature, config.Cfg.RewardCodeAuthorizedKeys)
+	_, err := verifySignatureAgainstKeys(req.Signature, config.Cfg.RewardCodeAuthorizedKeys)
 	if err != nil {
 		return fiber.NewError(fiber.StatusForbidden, "Unauthorized: "+err.Error())
 	}
 
+	// Generate a code
 	code, err := generateCode()
 	if err != nil {
 		return fiber.NewError(fiber.StatusInternalServerError, "Failed to generate code: "+err.Error())
 	}
 
+	var rewardAddress string
+
+	// Only create reward pool if deterministic secret is configured
+	if config.Cfg.LaunchpadDeterministicSecret != "" {
+		mintPubKey, err := solana.PublicKeyFromBase58(req.Mint)
+		if err != nil {
+			return fiber.NewError(fiber.StatusBadRequest, "Invalid mint address: "+err.Error())
+		}
+
+		claimAuthority, claimAuthorityPrivateKey, err := utils.DeriveEthAddressForMint(
+			[]byte("launchpad_reward_code"),
+			config.Cfg.LaunchpadDeterministicSecret,
+			mintPubKey,
+		)
+		if err != nil {
+			return fiber.NewError(fiber.StatusInternalServerError, "Failed to derive Ethereum key: "+err.Error())
+		}
+
+		// Convert the private key to the format expected by the SDK
+		privateKey, err := common.EthToEthKey(claimAuthorityPrivateKey)
+		if err != nil {
+			return fiber.NewError(fiber.StatusInternalServerError, "Failed to convert private key: "+err.Error())
+		}
+
+		// Create OpenAudio SDK instance and set the private key
+		oap := sdk.NewOpenAudioSDK(config.Cfg.AudiusdURL)
+		oap.SetPrivKey(privateKey)
```

Comment:
Oh so we do have to create a new sdk each time? :-(

Reply:
we do but i think it's cheap